### PR TITLE
JNG-5928 groups to views in jsl

### PR DESCRIPTION
**Pull request by noherczeg**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5928" title="JNG-5928" target="_blank">StoryJNG-5928</a>  Groups to views in JSL
  </td></table>
  <br />

### CODE REVIEW

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol [hunk 1]
```diff
@@ -93,5 +93,5 @@ operation JSL!TransferFieldDeclaration getTransferFieldDeclarationEquivalent():
     if (self.maps()) {
         return self.equivalent("CreateMappedTransferAttribute");
     }
-    throw "Could not determinate attribute type: " + s.fqName();
+    throw "Could not determinate attribute type: " + self.fqName();
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol:96]: determinate? Do you mean determine?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/ui/rowColumnDeclaration.eol [hunk 2]
```diff
@@ -11,5 +11,5 @@ operation JSL!UIRowColumnDeclaration getTransferFieldDeclarationEquivalent(): UI
     if (field.maps()) {
         return field.equivalent("CreateMappedTransferAttribute");
     }
-    throw "Could not determinate attribute type: " + s.fqName();
+    throw "Could not determinate attribute type: " + self.fqName();
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/ui/rowColumnDeclaration.eol:14]: determine?

**Review by bencegelei — APPROVED** [review]

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuLinkDeclarationViewPage.etl [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        rootMenu.equivalent("Application").pages.add(t);
+
+        t.dataElement = relation.equivalent("RelationType");
+
+        /*
+        log.info("===================");
+        log.info(t.name + ":");
+        log.info(relations.collect(r | r.name).concat(", "));
+        log.info("===================");
+        */
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuLinkDeclarationViewPage.etl:21]: Is it needed to be committed?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclarationFormPage.etl [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        for (table in relations.select(r | r.isKindOf(JSL!UIViewTableDeclaration))) {
+            var detailLink = table.getDetailLink();
+
+            if (detailLink.isDefined()) {
+                t.actions.add(table.equivalent("ViewTableDeclarationOpenPageAction"));
+            }
+        }
+        */
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclarationFormPage.etl:35]: Is it reserved for later use?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationViewPage.etl [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+        t.dataElement = s.transferRelation.target.equivalent("RelationType");
+
+        /*
+        log.info("===================");
+        log.info(t.name + ":");
+        log.info(relations.collect(r | r.name).concat(", "));
+        log.info("===================");
+        */
+
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationViewPage.etl:20]: Is it reserved for latrer use?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationFormPage.etl [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        for (table in s.getCreateFormModifier().form.getAllLinks()) {
+            var detailLink = table.getDetailLink();
+
+            if (detailLink.isDefined()) {
+                t.actions.add(table.equivalent("ViewTableDeclarationOpenPageAction"));
+            }
+        }
+*/
+
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationFormPage.etl:36]: Is it reserved for later use?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/rowDeclaration.etl [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+        t.onInit = s.equivalent("TableTableRefreshActionDefinition");
+
+        rootMenu.equivalent("Application").pageContainers.add(t);
+
+        /*
+        for (dataFeature in s.additionalMaskFeatures) {
+            var attributeType = dataFeature.getMember().mapAttributeType(s.eContainer.equivalent("ClassType"));
+            t.additionalMaskAttributes.add(attributeType);
+        }
+        */
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/rowDeclaration.etl:23]: IS it reserved for later use?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationViewPage.etl [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        for (table in relations.select(r | r.isKindOf(JSL!UIViewTableDeclaration))) {
+            var detailLink = table.getDetailLink();
+
+            if (detailLink.isDefined()) {
+                t.actions.add(table.equivalent("ViewTableDeclarationOpenPageAction"));
+            }
+        }
+        */
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationViewPage.etl:37]: Is it reserved for later use?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationAddSelectorPage.etl [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+                t.columns.add(col);
+                if (col.attributeType.isFilterable) {
+                   t.filters.add(link.equivalentDiscriminated("RowLinkDerivedColumnFilter", id));
+                }
+            } else {
+                // TODO generate open action from link with detail modifier
+            }
+
+        }
+*/
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationAddSelectorPage.etl:202]: IS it reserved for later use?

**Review by robertcsakany — APPROVED** [review]

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuLinkDeclarationViewPage.etl (hunk 3 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuLinkDeclarationViewPage.etl:21]: Used for occasional debugging.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclarationFormPage.etl (hunk 4 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclarationFormPage.etl:35]: Yes.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationFormPage.etl (hunk 6 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationFormPage.etl:36]: Yes.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationViewPage.etl (hunk 8 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationViewPage.etl:37]: Yes

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationAddSelectorPage.etl (hunk 9 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/menuTableDeclarationAddSelectorPage.etl:202]: Yes.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/rowDeclaration.etl (hunk 7 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/rowDeclaration.etl:23]: JSL doesn't have this yet, was copied from tatami-client, we'll need this once JSL gets the modifier.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationViewPage.etl (hunk 5 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclarationViewPage.etl:20]: Used for occasional debugging, once we have everything in place, it'll be removed.

**Pull request merged**